server/integrationtests/blocks_test.go

```diff
@@ -385,3 +386,139 @@ func TestGetSubtree(t *testing.T) {
 		require.Contains(t, blockIDs, childBlockID2)
 	})
 }
+
+func TestExportBlocks(t *testing.T) {
+	th := SetupTestHelper().InitBasic()
+	defer th.TearDown()
+
+	blocks, resp := th.Client.ExportBlocks("")
+	require.NoError(t, resp.Error)
+	initialCount := len(blocks)
+
+	parentBlockID1 := utils.NewID(utils.IDTypeBlock)
+	parentBlock1Title := fmt.Sprintf("Board 1 - %s", parentBlockID1)
+	parentBlockID2 := utils.NewID(utils.IDTypeBlock)
+	parentBlock2Title := fmt.Sprintf("Board 2 - %s", parentBlockID2)
+	childBlockID1 := utils.NewID(utils.IDTypeCard)
+	childBlock1Title := fmt.Sprintf("Card 1 - %s", childBlockID1)
+	childBlockID2 := utils.NewID(utils.IDTypeCard)
+	childBlock2Title := fmt.Sprintf("Card 2 - %s", childBlockID2)
+
+	newBlocks := []model.Block{
+		{
+			ID:       parentBlockID1,
+			RootID:   parentBlockID1,
+			CreateAt: 1,
+			UpdateAt: 1,
+			Title:    parentBlock1Title,
+			Type:     model.TypeBoard,
+		},
+		{
+			ID:       parentBlockID2,
+			RootID:   parentBlockID2,
+			CreateAt: 1,
+			UpdateAt: 1,
+			Title:    parentBlock2Title,
+			Type:     model.TypeBoard,
+		},
+		{
+			ID:       childBlockID1,
+			RootID:   parentBlockID1,
+			ParentID: parentBlockID1,
+			CreateAt: 1,
+			UpdateAt: 1,
+			Title:    childBlock1Title,
+			Type:     model.TypeCard,
+		},
+		{
+			ID:       childBlockID2,
+			RootID:   parentBlockID2,
+			ParentID: parentBlockID2,
+			CreateAt: 1,
+			UpdateAt: 1,
+			Title:    childBlock2Title,
+			Type:     model.TypeCard,
+		},
+	}
+	_, resp = th.Client.InsertBlocks(newBlocks)
+	require.NoError(t, resp.Error)
+
+	t.Run("export without root id", func(t *testing.T) {
+		blocks, resp = th.Client.ExportBlocks("")
+		require.NoError(t, resp.Error)
+		require.Len(t, blocks, initialCount+len(newBlocks))
+
+		blockTitles := make([]string, len(blocks))
+		for i, b := range blocks {
+			blockTitles[i] = b.Title
+		}
+		require.Contains(t, blockTitles, parentBlock1Title)
+		require.Contains(t, blockTitles, parentBlock2Title)
+		require.Contains(t, blockTitles, childBlock1Title)
+		require.Contains(t, blockTitles, childBlock2Title)
+	})
+
+	t.Run("export with root id", func(t *testing.T) {
+		t.Skip("TODO")
```

Comment:
Good evening @wiggin77 ! I have got a question here. I was trying to Insert some blocks so I could test the import  by using the new `root_id` param. But looking to the api code, it seems I can't get the blocks by using the blockIDs I have created on the test above (new blocks l407-442), and the reason is because some new IDs are being generated internally by the api: https://github.com/mattermost/focalboard/blob/main/server/api/api.go#L433 

In this scenario, how can I import a block that I have just created using the ID I used generated for them? (`utils.NewID(utils.IDTypeBlock)`) Because I won't have access to the ID generated by the API side.

Am I thinking right? 

Best regards,
Joao A.

Reply:
I agree @jpaldeano, that makes it a challenge to test from the client.  We need to change the block ID's because someone may export from one workspace and import into another, and if we don't change the block IDs they will collide.  

I suggest you create a block with a very unique title, and when validating you'll have to fetch all the blocks and find that unique title.  Bit of a hack but we do need to change the IDs on import.

Reply:
Thanks for clarifying that makes sense